index.js

```diff
@@ -86,6 +86,36 @@ function compatibleMessage(thrown, errMatcher) {
 }
 
 /**
+ * ### .getFunctionName(constructorFn)
+ *
+ * Returns the name of a function.
+ * This also includes a polyfill function if `constructorFn.name` is not defined.
+ *
+ * @name getFunctionName
+ * @param {Function} constructorFn
+ * @namespace Utils
+ * @api private
+ */
+
+function getFunctionName(constructorFn) {
+  var name = '';
+  if (typeof constructorFn.name === 'undefined') {
+    // Here we run a polyfill if constructorFn.name is not defined
+    var fnNamePrefixRegex = /^[\S\s]*?constructorFntion\s*/;
```

Comment:
These regexes are a little redundant; `[\S\s]` will match any non-whitespace character, plus any whitespace character - so in other words... any character. In addition the `?` following the `*` is redundant. I also think using `replace` to strip out parts is a little cumbersome, when we can use `match` to extract the right bits. So something like:

```js
String(function foo(){}).match(/\s*function\s*([^\(]+)/)[1]
```

However I did a little testing, and it seems that IE9, at least, preserves comments in functions, so we should also look for those and strip them out where necessary. So something a bit more robust would be:

```js
String(function/*one*/ foo/*two*/(){}).match(/\s*function(?:\s*|\s*\/\*[^(?:*/)]+\*\/\s*)*?([^\(\/]+)/)[1]
```

Also (small aside) think it'd be worth caching the regexp as a var outside of the function - so it isn't instantiated each time. So ultimately this whole block becomes:

```js
var functionNameMatch = /\s*function(?:\s*|\s*\/\*[^(?:*/)]+\*\/\s*)*?([^\(\/]+)/;

function getFunctionName(constructor) {
  if (typeof constructor.name === 'undefined') {
    var match = String(constructor).match(functionNameMatch);
    if (match) {
      return match[1];
    }
    return '';
  }
  return constructor.name;
}
```

Of course it's probably wise to also add some unit tests which demonstrate that it successfully extracts the right name even with comments inline.